### ISSUE TITLE
Missing data when using dynamic routing

### ISSUE THREAD

**Issue by benjaminrancourt** [TryGhost/Ghost#14206]
### Issue Summary

Hi,

First, thank you for developing Ghost, it's one of my favourite applications!

When using **dynamic routing**, some data seems missing from the page, like the Ghost header or Ghost footer.

### Steps to Reproduce

1. Create (or use) a new page. like `/about/`.
2. Fill all fields that Ghost provides inside the settings of the page (metadata, Twitter, Facebook, Code injection).
3. Publish it.
4. Navigate to the newly published page and save the HTML output (**OUTPUT_1**).
5. In the backend, create a new route in the routes.yaml file.
```yaml
routes:
  /fr/a-propos/:
    data: page.about
    template: page

collections:
  /:
    permalink: /{slug}/
    template: index

taxonomies:
  tag: /tag/{slug}/
  author: /author/{slug}/
```
6. Go to the new URL (`/fr/a-propos/`) or the previous one (`/about`).
7. Save the HTML output (**OUTPUT_2**) and notice that our page is now broken.
8. Inside the theme, edit the `post.hbs` file to change its context from `{{#post}}` to `{{#page}}`.
9. Go back to the published page and save for the last time the HTML output (**OUTPUT_3**).

For **OUTPUT_2** and **OUTPUT_3**, I was expecting the same result as **OUTPUT_1**.

Summarily, it's currently missing:
* The canonical URL is overridden by the URL inside the `routes.yaml` file.
  * `link rel="canonical"`
  * `meta property="og:url"`
  * `meta name="twitter:url"`
* The `meta property="article:published_time"` has been removed.
* The `meta property="article:tag"` has been removed.
* Twitter labels and data have been removed.
* The `ghost_head` and `ghost_footer` is not injected anymore.
* Some properties are missing from the `application/ld+json` schema.

### Ghost Version

4.36

### Node.js Version

LTS

### How did you install Ghost?

Ghost PRO

### Database type

MySQL 8

### Code of Conduct

- [X] I agree to be friendly and polite to people in this repository

**Comment by benjaminrancourt** [comment]
Here are screenshots of the 3 outputs from my steps.

## OUTPUT_1
![1](https://user-images.githubusercontent.com/7907713/155383618-d0cc42e9-efa7-4c7d-86c0-6d1ab0b0b0e9.jpg)

## OUTPUT_2
![2](https://user-images.githubusercontent.com/7907713/155383629-cf0591f9-40c2-4c9d-8724-17e7e8e3af4a.jpg)

## OUTPUT_3
![3](https://user-images.githubusercontent.com/7907713/155383640-8bacdbd2-bf08-4d17-9e9c-d676df37cd18.jpg)
Note: I probably didn't capture the header, but the visual is identical to the first output.

**Comment by benjaminrancourt** [comment]
And finally, here are the differences in the HTML between the output.

## Between OUTPUT_1 and OUTPUT_2
https://www.diffchecker.com/NklejbuD

![image](https://user-images.githubusercontent.com/7907713/155384336-27fd65c9-3803-4a40-a9ac-fed69972ac1a.png)
![image](https://user-images.githubusercontent.com/7907713/155384388-edff6c9d-8067-4fbc-8a4c-14d7667685b9.png)
![image](https://user-images.githubusercontent.com/7907713/155384458-aaffd519-466d-43f5-8c34-1f69975c0472.png)
![image](https://user-images.githubusercontent.com/7907713/155384497-953bae15-72d6-4bd5-b5b9-2846439a1470.png)


## Between OUTPUT_2 and OUTPUT_3
https://www.diffchecker.com/AGTDYDvz

## Between OUTPUT_1 and OUTPUT_3
https://www.diffchecker.com/kWNEO0wf

![image](https://user-images.githubusercontent.com/7907713/155384590-61331095-403f-417b-88a9-b3a55d80b093.png)
![image](https://user-images.githubusercontent.com/7907713/155384637-ae2135d8-1917-40a6-9bd7-eff3b2a39b09.png)
![image](https://user-images.githubusercontent.com/7907713/155384675-d386ee09-4148-44fd-8f90-5f4641680619.png)
![image](https://user-images.githubusercontent.com/7907713/155384744-9f670abe-a374-44cd-8692-97dabf97629d.png)

**Comment by ErisDS** [comment]
@benjaminrancourt do you still see this on Ghost v5?

**Comment by ErisDS** [comment]
I checked and still see this in v5. It's essentially the same as this old chestnut: https://github.com/TryGhost/Ghost/issues/10042

I have found a little fix.

**Comment by benjaminrancourt** [comment]
Thanks @ErisDS, you were faster than me to test it! 

I just get the latest Docker image available (v5.1.1) and I confirm that your commit fixed the bug!

Thanks again!
